src/common/pf_dcp.c

```diff
@@ -204,9 +205,57 @@ static void pf_dcp_responder (pnet_t * net, void * arg, uint32_t current_time)
  */
 static void pf_dcp_clear_sam (pnet_t * net, void * arg, uint32_t current_time)
 {
+   LOG_DEBUG (PF_DCP_LOG, "DCP(%d): SAM timeout, clear stored mac\n", __LINE__);
    net->dcp_sam = mac_nil;
 }
 
+/**
+ * @internal
+ * Reset SAM timeout
+ * SAM limits the communication to a single remote MAC address
+ * for 3 seconds.
+ *
+ * @param net              InOut: The p-net stack instance
+ * @param mac              In: Remote MAC address
+ */
+static void pf_dcp_reset_sam_timeout (pnet_t * net, const pnet_ethaddr_t * mac)
+{
+   /* Make sure no other MAC address is used in the DCP communication
+    * for 3 seconds */
+   LOG_DEBUG (
+      PF_DCP_LOG,
+      "DCP(%d): Update SAM mac and reset timeout\n",
+      __LINE__);
+
+   memcpy (&net->dcp_sam, mac, sizeof (net->dcp_sam));
+   (void)pf_scheduler_remove (net, dcp_sync_name, net->dcp_sam_timeout);
+   net->dcp_sam_timeout = 0;
+   (void)pf_scheduler_add (
+      net,
+      3 * 1000 * 1000, /* 3s */
```

Comment:
I think we should have a define for this.

Reply:
Done